002: 認証セッションモジュールの実装

Title: [URGENT] @/lib/auth/sessionモジュールの実装
Labels: bug, high-priority, blocking, auth, claude-code
Assignee: （空欄または利用可能なassignee）

Body:
## 🚨 緊急度：最高

### 問題の概要
`src/app/api/checkout/route.ts`から参照されている`@/lib/auth/session`モジュールが存在せず、ビルドが失敗しています。

### 具体的なエラー


### 具体的なエラー
./src/app/api/checkout/route.ts
Module not found: Can't resolve '@/lib/auth/session'

### 必要な実装

#### 作成するファイル
**ファイル名**: `src/lib/auth/session.ts`

#### 実装が必要な関数
```typescript
// 必要な関数シグネチャ例
export async function getServerSession(): Promise<Session | null> {
  // Supabase認証との連携
  // サーバーサイドでの認証情報取得
}

export interface Session {
  user: {
    id: string;
    email: string;
    // その他必要な属性
  };
  // その他セッション情報
}
参考にするファイル

src/components/providers/auth-provider.tsx - 既存の認証実装
Next.js App Router認証パターン
Supabaseサーバーサイド認証ドキュメント

技術要件

Next.js App RouterのServer Componentから利用可能
Supabaseの認証情報と連携
TypeScript型定義の完備
エラーハンドリングの実装

成功条件

 src/lib/auth/session.ts ファイル作成完了
 getServerSession 関数実装完了
 checkout APIルートのビルドエラー解消
 TypeScript型エラー解消

影響範囲
このモジュールの不足により、決済機能全体が利用できない状態です。

---

### **Issue #003: useRealtimeAuctionフック循環参照修正**

**GitHub Issue作成内容：**

--- a/src/lib/auth/session.ts
+++ b/src/lib/auth/session.ts
@@ -1,0 +1,47 @@
+import { cookies } from 'next/headers';
+import { createClient } from '@/lib/supabase/server';
+
+export interface Session {
+  user: {
+    id: string;
+    email: string;
+    name?: string;
+  };
+}
+
+export async function getServerSession(): Promise<Session | null> {
+  try {
+    const cookieStore = await cookies();
+    const supabase = createClient(cookieStore);
+    
+    const { data: { user }, error } = await supabase.auth.getUser();
+    
+    if (error || !user) {
+      return null;
+    }
+    
+    // ユーザープロファイルから名前を取得
+    const { data: profile } = await supabase
+      .from('profiles')
+      .select('display_name, username')
+      .eq('id', user.id)
+      .single();
+    
+    // 名前の優先順位: display_name > username > user_metadata.username > emailの@前
+    const name = profile?.display_name || 
+                 profile?.username || 
+                 user.user_metadata?.username ||
+                 user.email?.split('@')[0];
+    
+    return {
+      user: {
+        id: user.id,
+        email: user.email || '',
+        name: name || undefined,
+      }
+    };
+  } catch (error) {
+    console.error('getServerSession error:', error);
+    return null;
+  }
+}

